sdk/src/main/java/com/bugsnag/android/Error.java

```diff
@@ -399,8 +399,13 @@ HandledState getHandledState() {
         @HandledState.SeverityReason
         private String severityReasonType;
 
-        Builder(@NonNull Configuration config, @NonNull Throwable exception, Session session) {
-            this.threadState = new ThreadState(config);
+        Builder(@NonNull Configuration config,
+                @NonNull Throwable exception,
+                @Nullable Session session,
+                @NonNull Thread thread,
+                boolean unhandled) {
+            Throwable exc = unhandled ? exception : null;
+            this.threadState = new ThreadState(config, thread, Thread.getAllStackTraces(), exc);
```

Comment:
If an exception is passed in (default for unhandled errors), the exception stacktrace will be serialised rather than the thread stacktrace